core/trino-main/src/main/java/io/trino/operator/TableMutationOperator.java

```diff
@@ -25,41 +22,45 @@
 
 import java.util.List;
 import java.util.OptionalLong;
+import java.util.function.Supplier;
 
 import static com.google.common.base.Preconditions.checkState;
 import static io.trino.spi.type.BigintType.BIGINT;
 import static java.util.Objects.requireNonNull;
 
-public class TableDeleteOperator
+public class TableMutationOperator
         implements Operator
 {
     public static final List<Type> TYPES = ImmutableList.of(BIGINT);
 
-    public static class TableDeleteOperatorFactory
+    private final OperatorContext operatorContext;
+    private final Supplier<OptionalLong> executeOperation;
+    private boolean finished;
+
+    public static class TableMutationOperatorFactory
             implements OperatorFactory
     {
         private final int operatorId;
         private final PlanNodeId planNodeId;
-        private final Metadata metadata;
-        private final Session session;
-        private final TableHandle tableHandle;
+        private final Supplier<OptionalLong> executeOperation;
         private boolean closed;
 
-        public TableDeleteOperatorFactory(int operatorId, PlanNodeId planNodeId, Metadata metadata, Session session, TableHandle tableHandle)
+        public TableMutationOperatorFactory(
+                int operatorId,
+                PlanNodeId planNodeId,
+                Supplier<OptionalLong> executeOperation)
```

Comment:
`Supplier` is generally expected to be free of side effects, so it is not a good fit for this purpose. Add an interface in this class with a single method. E.g.,

```java
interface Operation
{
    OptionalLong execute();
}
```